src/one.rs

```diff
@@ -75,8 +77,20 @@ enum WriteStorage<'w, Trait: ?Sized> {
 /// [`WorldQuery`] adapter that fetches entities with exactly one component implementing a trait.
 pub struct One<T>(pub T);
 
+/// [`WorldQuery`] adapter that fetches entities with exactly one component implementing a trait,
+/// with the condition that the component must also have changed since the last tick
+pub struct ChangedOne<T>(pub T);
+
+/// [`WorldQuery`] adapter that fetches entities with exactly one component implementing a trait,
+/// with the condition that the component was newly added since the last tick
+pub struct AddedOne<T>(pub T);
```

Comment:
Just making sure, this essentially a shorthand for `Query<One<&dyn Trait>, AddedOneFilter<dyn Trait>>`, right?